ci/docker/conda-python-pandas.dockerfile

```diff
@@ -34,3 +34,5 @@ RUN mamba install -q -y --file arrow/ci/conda_env_sphinx.txt && \
 COPY ci/scripts/install_pandas.sh /arrow/ci/scripts/
 RUN mamba uninstall -q -y numpy && \
     /arrow/ci/scripts/install_pandas.sh ${pandas} ${numpy}
+
+RUN apt-get update && apt-get install -y patchelf
```

Comment:
I'm curious, why is this needed?

Reply:
When either meson or meson-python creates a wheel file the "installation" process will bundle the required libraries and change the linkage to using rpath resolution. I believe similar to auditwheel, but it is handled entirely by meson/meson-python.

 @eli-schwartz can speak better to the inner workings

Reply:
would be good to understand this in detail, how are the symbols updated for example, if what we want to do in the future is split the wheels into different components, similar to what we did for conda

Reply:
For pyprojects that have meson.build files which compile and install BOTH a shared library, and a python extension that links to the shared library, meson-python can internally package it in a wheel using the same tricks that auditwheel uses to bundle up completely third-party shared libraries.

Namely, meson-python installs the shared library into a private directory in site-packages (.xxx.mesonpy.libs/) and uses patchelf to modify the extension module to search there for its shared library dependencies.

You shouldn't need patchelf if that relocation isn't being done. Users of auditwheel are probably accustomed to installing patchelf anyway, though. :)